src/profile/request/extra_data/third_party/attribute.issuance.details.builder.js

```diff
@@ -0,0 +1,59 @@
+const { Validation } = require('../../../../util');
+const SandboxIssuingAttributes = require('./issuing.attributes');
+const SandboxDefinition = require('./definition');
+const SandboxAttributeIssuanceDetails = require('./attribute.issuance.details');
+const SandboxAttributeIssuanceDetailsValue = require('./attribute.issuance.details.value');
+
+class SandboxAttributeIssuanceDetailsBuilder {
+  constructor() {
+    this.definitions = [];
+  }
+
+  /**
+   * @param {string} issuanceToken
+   *
+   * @returns {this}
+   */
+  withIssuanceToken(issuanceToken) {
+    Validation.notNullOrEmpty(issuanceToken, 'issuanceToken');
+    Validation.isString(issuanceToken, 'issuanceToken');
+    this.issuanceToken = issuanceToken;
+    return this;
+  }
+
+  /**
+   * @param {Date} expiryDate
+   *
+   * @returns {this}
+   */
+  withExpiryDate(expiryDate) {
+    Validation.instanceOf(expiryDate, Date, 'expiryDate');
+    this.expiryDate = expiryDate;
+    return this;
+  }
+
+  /**
+   * @param {string} definition
+   *
+   * @returns {this}
+   */
+  withDefinition(definition) {
```

Comment:
I agree with only having a builder method which accepts the string, but do you think we should call it `WithDefinitionName`? So that for languages where we can't have method overloading, that we could use `WithDefinition` to take the `AttributeDefinition` object in future

Reply:
I would have thought we will add extra parameters to this method, rather than allow the object to be passed in?

Reply:
Yep, let's do that 👍